streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -663,6 +663,12 @@ public class StreamsConfig extends AbstractConfig {
     public static final String TOPOLOGY_OPTIMIZATION_CONFIG = "topology.optimization";
     private static final String TOPOLOGY_OPTIMIZATION_DOC = "A configuration telling Kafka Streams if it should optimize the topology, disabled by default";
 
+    public static final String SELF_JOIN_OPTIMIZATION_CONFIG = "self.join.optimization";
```

Comment:
I still have some concerns about the extra configs pattern here, but I will leave for open discuss in the KIP thread.

Reply:
I agree with your suggestion in the KIP and will make the changes.